app/models/knight.rb

```diff
@@ -1,9 +1,26 @@
 class Knight < ChessPiece
+
   def display
     if color == "black"
       return "&#9822;"
     else
       return "&#9816;"
     end
   end
+
+  def is_valid_move?(board, stop)
+    (is_horizontal_move?(board, stop) || is_vertical_move?(board, stop)) && (!chess_piece || chess_piece.color != color)
```

Comment:
What does the second half of this statement do? I think you're checking if the space is occupied and if its the opponent, but I'm not positive. You can use the is_space_occupied? and is_opponent? methods if that makes it easier.

Reply:
agreed.